docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,196 @@
+This means that when containers share a non-user namespace they must also
+share a user namespace whether that be a private user namespace or
+the host's user namespace.  
+* Currently you may only specify a single user namespace remap value
+to the docker daemon.  This means you cannot ensure namespaces are
+using different UIDs and GIDs for remapping, they will all be remapped
+to the same value(s).
+* [Known restrictions when running in a remap environment](https://docs.docker.com/engine/reference/commandline/dockerd/#user-namespace-known-restrictions)
+* A cluster may be running with some nodes performing remapping and 
+some not.  Any runtime behavior should be isolated to the kubelet
+and information about remapping should come from the CRI.
+
+### Detecting and Driving Private User Namespaces Behavior
+
+In order to Do The Right Thing in any solution the system must be able 
+to detect that private user namespaces are in use.  
+
+In Docker, this is a daemon setting and is [exposed via 
+the info endpoint](https://github.com/docker/docker/commit/ae74092e450f1f2665b90257b65513cc0c19702f).
+This information is available in Docker 1.13+ only.
+
+However, in rkt this is a run parameter so it is
+unknown if private user namespaces will be used until the container launch
+command is created.  
+
+As a consistent and backwards compatible solution to both driving any behavior to escape 
+user namespaces (see system defaulting below) and to also control
+the use of `--private-users` in the rkt `run` command a feature
+gate can be added to the Kubelet as was done in the [experimental
+support](https://github.com/kubernetes/kubernetes/pull/31169/files#diff-c5e4440d8576a91980ab36fe63556326).
+
+### Volumes
+
+The system must ensure that the remapped UID and GID have
+access to the correct directories under `/var/lib/kubelet/pods` in order
+to perform mounting.  Currently the Kubelet root directory is created
+with 0750 permissions and are owned by root:root.  Since pods are running
+with at least the root group (`runAsUser` may be specified as well as 
+supplemental groups) the pod has access to read these directories.  In
+a remap environment the group itself may change so the pod will no longer
+be able to access these directories.  In order to fix this the root 
+directory should be given 0755 access in the Kubelet's `setupDataDirs`.
+
+Beneath the Kubelet root directory each pod is given it's own directory
+named by a unique identifier containing:
+
+* containers
+* plugins
+* volumes
+
+Rather than increasing the access on directories under the root directory,
+the pod directory (and sub directories) may be owned by the remapped
+GID.  This is very similar to the `fsGroup`, however `fsGroup` is not applied
+to items like the termination logs.
+
+How this GID is retrieved should be a container runtime detail.  This
+leaves the container runtimes free to manage the permissions of the 
```

Comment:
thoughts on allowing each runtime to managed the gid ownership of the pod root directory and below?